tensorflow/contrib/lite/nnapi_delegate.cc

```diff
@@ -258,11 +258,11 @@ TfLiteStatus AddOpsAndParams(
       augmented_inputs.push_back(next_id++);
     };
 
-    auto add_vector_int32 = [&](const int* values, uint32_t num_values) {
+    auto add_vector_int32 = [&](const uint32_t* values, uint32_t num_values) {
       ANeuralNetworksOperandType operand_type{
           .type = ANEURALNETWORKS_TENSOR_INT32,
           .dimensionCount = 1,
-          .dimensions = &num_values};
+          .dimensions = values};
```

Comment:
I don't think this is right. Here we're describing the *shape* of the (1D) vector, not the actual vector payload (which happens to be a shape for squeeze). So the dimension argument should be a single element which describes the length fo the vector, which is num_values.

Have you actually encountered a segfault here?

Reply:
Thanks Jared, @jdduke . Yes, I encountered a segfault when my NNAPI calls squeeze operation. Actually, operand_type.dimensions[0] is equal num_values and it is also equal to values[0], because vector is 1-D tensor. But the num_values is just a local variable,  it will be removed when the function call is finished.  Then, the .dimensions pointer will point to other address. I think that's the reason I met segfault here. So .dimensions is better to equal values or use a reference for num_values. Am I correct?

Reply:
Hmm, I'm pretty sure that calling addOperand() will internally trigger a copy of the provided dimension data. Can you add any more details about the segfault you're seeing? I'm wondering if the squeeze dimensions are actually empty in this case? Can you check?

Reply:
Thanks for your information, @jdduke . I think I got the point.  I directly use the the dimensions pointer transfer into addOperand() without copying the dimension values in our NNAPI implementation. That's why I had a segfault. You are correct! Thank you for your help.  I will close the request, since it is not necessary to have any changes.